docs/dev/RELEASE.md

```diff
@@ -0,0 +1,87 @@
+## Introduction
+This document guides maintainers through the process of creating an Envoy Gateway release.
+
+## Prerequisites
+- Permissions to push to the Envoy Gateway repository.
+
+## Creating a Minor Release
+
+1. Clone the repo, checkout the `main` branch, ensure it’s up-to-date, and your local branch is clean.
+2. Create a changelog corresponding to the release number that summarizes the changes included in the release.
+   Reference previous [changelogs][changelogs] for additional details.
+3. Submit a [Pull Request][pr] to merge the changelog into the main branch. This should be the last commit to main
+   before cutting the release.
+4. Create a new release branch from `main`. The release branch should be named
+   `release/v${MAJOR_VERSION}.${MINOR_VERSION}.0`, e.g. `release/v0.3.0`.
+   ```shell
+   git checkout -b release/v0.3.0
+   ```
+5. Push the branch to the Envoy Gateway repo.
+6. Create a topic branch and update the release tag references in the [Quickstart Guide][quickstart].
+7. Sign, commit, and push your changes to your fork. Send a PR to get your changes merged to the release branch.
+   Do not proceed until your PR is merged into the release branch.
+8. Confirm that the [release workflow][release_workflow] for your PR completed successfully.
```

Comment:
markdown nit: If you'd just written the reference `release workflow: https://…`, then here you could just write `[release workflow][]` without having to stutter.

Reply:
Thanks for the pro tip!